### PR TITLE
Add some PF configure cross tests

### PR DESCRIPTION
**Pull request by iwahbe**
This PR is best reviewed commit by commit.

### CODE REVIEW

File: pkg/pf/tests/provider_configure_test.go [hunk 1]
```diff
@@ -30,27 +30,77 @@ import (
...
+}
 
-	t.Run("secret-string", crosstests.MakeConfigure(
+type testConfigurePrimitive struct {
+	zeroValue, nonZeroValue    cty.Value
+	attrOptional, attrRequired schema.Attribute
+}
+
+func (tc testConfigurePrimitive) run(t *testing.T) {
+	t.Parallel()
```
Comment (VenelinMartinov) [inline pkg/pf/tests/provider_configure_test.go:311]: I think this needs t.Helper() to make the stack trace for the failure point to the actual parameters

**Review by VenelinMartinov — COMMENTED** [review]
I don't see any tests for Block types - are these not allowed in the config?

File: pkg/pf/tests/internal/cross-tests/util.go [hunk 2]
```diff
@@ -50,37 +49,6 @@ func (s testLogSink) log(kind string, sev diag.Severity, urn resource.URN, msg s
 	return nil
 }
 
-func convertResourceValue(t *testing.T, properties resource.PropertyMap) map[string]any {
```
Comment (t0yv0) [inline pkg/pf/tests/internal/cross-tests/util.go:53]: Nice!

File: pkg/pf/tests/provider_configure_test.go [hunk 3]
```diff
@@ -59,20 +109,60 @@ func TestConfigure(t *testing.T) {
 func TestConfigureInvalidTypes(t *testing.T) {
 	t.Setenv("PULUMI_DEBUG_YAML_DISABLE_TYPE_CHECKING", "true")
 
-	t.Run("bool-type-conversion", crosstests.MakeConfigure(
+	t.Run("string-as-bool", crosstests.MakeConfigure(
```
Comment (t0yv0) [inline pkg/pf/tests/provider_configure_test.go:345]: Yet another dimension maybe to worry about, but we recently had a problem here around requesting the type to change at Pulumi level (skipMetadataAPiCheck). Perhaps another PR. https://github.com/pulumi/pulumi-terraform-bridge/issues/2520

**Review by t0yv0 — APPROVED** [review]
Nice!!

**Comment by t0yv0** [comment]
Block types would be nice to cover TBH, perhaps another PR.

**Comment by t0yv0** [comment]
Codecov indicates missing coverage:

```
+ func ctyToTftypes(v cty.Value) tftypes.Value {
+ 	typ := v.Type()
+ 	if !v.IsKnown() {
+ 		return tftypes.NewValue(ctyTypeToTfType(typ), tftypes.UnknownValue)
+ 	}
```

This is out of scope for this PR but I remember being burned by making changes that broke handling of unknowns through provider configuration. We should probably cover that, WDYT?

**Comment by iwahbe** [comment]
> Codecov indicates missing coverage:
> 
> ```
> + func ctyToTftypes(v cty.Value) tftypes.Value {
> + 	typ := v.Type()
> + 	if !v.IsKnown() {
> + 		return tftypes.NewValue(ctyTypeToTfType(typ), tftypes.UnknownValue)
> + 	}
> ```
> 
> This is out of scope for this PR but I remember being burned by making changes that broke handling of unknowns through provider configuration. We should probably cover that, WDYT?

We definitely should, and it should be included in the [testing epic for Configure](https://github.com/pulumi/home/issues/3719). We don't have support for unknowns in cross-tests now: https://github.com/pulumi/pulumi-terraform-bridge/issues/1858.

**Comment by iwahbe** [comment]
> Block types would be nice to cover TBH, perhaps another PR.

I added a simple test, and found a bug: https://github.com/pulumi/pulumi-terraform-bridge/issues/2564. This PR is meant to stand up some coverage, not complete coverage.

**Pull request merged**